fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/mixin/event/interaction/client/MinecraftClientMixin.java

```diff
@@ -157,4 +162,25 @@ private void injectUseEntityCallback(CallbackInfo ci, Hand[] hands, int i1, int
 			ci.cancel();
 		}
 	}
+
+	@Inject(
+			method = "doAttack",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/client/network/ClientPlayerEntity;getStackInHand(Lnet/minecraft/util/Hand;)Lnet/minecraft/item/ItemStack;"
+			),
+			cancellable = true
+	)
+	private void onDoAttack(CallbackInfoReturnable<Boolean> cir) {
+		ActionResult actionResult = ClientPreAttackCallback.EVENT.invoker().onClientPlayerPreAttack(MinecraftClient.getInstance().player);
+
+		if (actionResult != ActionResult.PASS) {
+			if (actionResult.isAccepted()) {
+				attackCooldown = 10;
```

Comment:
Not a massive fan of the magic numbers, MinecraftClient.field_32145 is likely this constant? Might be nice to get it mapped in yarn?